src/scalar.rs

```diff
@@ -1231,3 +1265,21 @@ fn test_double() {
 
     assert_eq!(a.double(), a + a);
 }
+
+#[test]
+fn test_ord() {
+    assert!(Scalar::one() > Scalar::zero());
```

Comment:
Add a test case that fails if `Ord` is implemented using the Montgomery representation:
```suggestion
    assert!(Scalar::one() > Scalar::zero());

    // Check that Ord is not comparing the Montgomery representations:
    // 4R mod q = 0x6092c566b31415be66313fbfb2f13fd56212dfe8000d200800000007fffffff8
    // 5R mod q = 0x04c9cf6d363b9de5cc83b7a7960bb7c566d9f3df00120c0b0000000afffffff5
    assert!(Scalar::from(5) > Scalar::from(4));

```

Reply:
Done in b5ef8e2